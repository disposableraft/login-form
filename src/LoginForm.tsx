import React, { FunctionComponent, useState } from "react";
import { Formik, Form, FormikHelpers } from "formik";
import TextInput from "./TextInput";

type Inputs = {
  firstName: string;
  lastName: string;
  email: string;
  password: string;
};

const LoginForm: FunctionComponent = () => {
  const onSubmit = (values: Inputs, setSubmitting: any) => {
    console.log(values);
  };

  return (
    <div>
      <Formik
        initialValues={{
          firstName: "",
          lastName: "",
          email: "",
          password: "",
        }}
        onSubmit={(
          values: Inputs,
          { setSubmitting }: FormikHelpers<Inputs>
        ) => {
          onSubmit(values, setSubmitting);
        }}
        validate={(values) => {
          let errors: any = {};
          if (!values.firstName) {
            errors.firstName = "Required";
          }
          if (!values.lastName) {
            errors.lastName = "Required";
          }
          if (!values.email) {
            errors.email = "Required";
          }
          if (!values.password) {
            errors.password = "Required";
          }
          return errors;
        }}
      > 
        <Form>
          <TextInput name="firstName" type="text" label="First Name" />
          <TextInput name="lastName" type="text" label="Last Name" />
          <TextInput name="email" type="email" label="Email" />
          <TextInput name="password" type="password" label="Password" />
          <button data-testid="submit" type="submit" disabled={false}>
            Register
          </button>
        </Form>
      </Formik>
    </div>
  );
};

export default LoginForm;
