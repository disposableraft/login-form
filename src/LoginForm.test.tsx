import React from "react";
import LoginForm from "./LoginForm";
import { render, fireEvent } from "@testing-library/react";

describe("LoginForm", () => {
  test("Matches snapshot without values", () => {
    const { container } = render(<LoginForm />);
    expect(container).toMatchSnapshot();
  });

  test("Matches snapshot with values", async () => {
    const { findByTestId, getByLabelText } = render(<LoginForm />);
    const fields = ["First Name", "Last Name", "Email", "Password"];
    fields.forEach((name) => {
      const input = getByLabelText(name);
      fireEvent.change(input, { target: { value: "May you be happy" } });
    });
    const form = await findByTestId('form-login')
    expect(form).toMatchSnapshot();
  });
});
