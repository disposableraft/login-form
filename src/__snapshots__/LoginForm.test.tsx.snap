// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Matches snapshot with values 1`] = `
<div
  class="css-0"
  data-testid="form-login"
>
  <form
    action="#"
  >
    <div>
      <label
        for="firstName"
      >
        First Name
      </label>
      <input
        id="firstName"
        name="firstName"
        type="text"
        value="May you be happy"
      />
    </div>
    <div>
      <label
        for="lastName"
      >
        Last Name
      </label>
      <input
        id="lastName"
        name="lastName"
        type="text"
        value="May you be happy"
      />
    </div>
    <div>
      <label
        for="email"
      >
        Email
      </label>
      <input
        id="email"
        name="email"
        type="email"
        value="May you be happy"
      />
    </div>
    <div>
      <label
        for="password"
      >
        Password
      </label>
      <input
        id="password"
        name="password"
        type="password"
        value="May you be happy"
      />
    </div>
  </form>
</div>
`;

exports[`LoginForm Matches snapshot without values 1`] = `
<div>
  <div
    class="css-0"
    data-testid="form-login"
  >
    <form
      action="#"
    >
      <div>
        <label
          for="firstName"
        >
          First Name
        </label>
        <input
          id="firstName"
          name="firstName"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="lastName"
        >
          Last Name
        </label>
        <input
          id="lastName"
          name="lastName"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="email"
        >
          Email
        </label>
        <input
          id="email"
          name="email"
          type="email"
          value=""
        />
      </div>
      <div>
        <label
          for="password"
        >
          Password
        </label>
        <input
          id="password"
          name="password"
          type="password"
          value=""
        />
      </div>
    </form>
  </div>
</div>
`;
